incubator/xray/README.md

```diff
@@ -0,0 +1,222 @@
+$ helm upgrade --set common.xrayVersion=2.1.0 incubator/xray
+```
+
+## Remove
+Removing a **helm** release is done with
+```bash
+# Remove the Xray services and data tools
+$ helm delete --purge xray
+
+# Remove the data disks
+$ kubectl delete pvc -l release=xray
+```
+
+### Create a unique Master Key
+JFrog Xray requires a unique master key to be used by all micro-services in the same cluster. By default the chart has one set in values.yaml (`common.masterKey`).
+
+**This key is for demo purpose and should not be used in a production environment!**
+
+You should generate a unique one and pass it to the template at install/upgrade time.
+```bash
+# Create a key
+$ export MASTER_KEY=$(openssl rand -hex 32)
+$ echo ${MASTER_KEY}
+
+# Pass the created master key to helm
+$ helm install --set common.masterKey=${MASTER_KEY} -n xray incubator/xray
+```
+**NOTE:** Make sure to pass the same master key with `--set common.masterKey=${MASTER_KEY}` on all future calls to `helm install` and `helm upgrade`!
+
+## Special deployments
+This is a list of special use cases for non-standard deployments
+
+### High Availability
+For **high availability** of Xray, just need to set the replica count per pod be equal or higher than **2**. Recommended is **3**.
+> It is highly recommended to also set **RabbitMQ** to run as an HA cluster.
+```bash
+# Start Xray with 3 replicas per service and 3 replicas for RabbitMQ
+$ helm install -n xray --set replicaCount=3,rabbitmq-ha.replicaCount=3 incubator/xray
+```
+
+### External Databases
+There is an option to use external database services (MongoDB or PostgreSQL) for your Xray.
+
+#### MongoDB
+To use an external **MongoDB**, You need to set Xray **MongoDB** connection URL.
+
+For this, pass the parameter: `global.mongoUrl=${XRAY_MONGODB_CONN_URL}`.
+
+**IMPORTANT:** Make sure the DB is already created before deploying Xray services
+```bash
+# Passing a custom MongoDB to Xray
+
+# Example
+# MongoDB host: custom-mongodb.local
+# MongoDB port: 27017
+# MongoDB user: xray
+# MongoDB password: password1_X
+
+$ export XRAY_MONGODB_CONN_URL='mongodb://xray:password1_X@custom-mongodb.local:27017/?authSource=xray&authMechanism=SCRAM-SHA-1'
+$ helm install -n xray --set global.mongoUrl=${XRAY_MONGODB_CONN_URL} incubator/xray
+```
+
+#### PostgreSQL
+To use an external **PostgreSQL**, You need to disable the use of the bundled **PostgreSQL** and set a custom **PostgreSQL** connection URL.
+
+For this, pass the parameters: `postgresql.enabled=false` and `global.postgresqlUrl=${XRAY_POSTGRESQL_CONN_URL}`.
+
+**IMPORTANT:** Make sure the DB is already created before deploying Xray services
+```bash
+# Passing a custom PostgreSQL to Xray
+
+# Example
+# PostgreSQL host: custom-postgresql.local
+# PostgreSQL port: 5432
+# PostgreSQL user: xray
+# PostgreSQL password: password2_X
+
+$ export XRAY_POSTGRESQL_CONN_URL='postgres://xray:password2_X@custom-postgresql.local:5432/xraydb?sslmode=disable'
```

Comment:
Don't make the user specify the password in plain text. Instead the user should have to specify a secret that contains the URL.